mono/metadata/object-internals.h

```diff
@@ -631,7 +631,9 @@ typedef struct {
 	gpointer (*create_delegate_trampoline) (MonoDomain *domain, MonoClass *klass);
 	gpointer (*interp_get_remoting_invoke) (gpointer imethod, MonoError *error);
 	GHashTable *(*get_weak_field_indexes) (MonoImage *image);
-	void     (*runtime_telemetry_callback) (void);
+#ifdef HOST_DARWIN
```

Comment:
It's `TARGET_OSX` instead of `HOST_DARWIN`, otherwise it will impact iOS as well. Please change it *everywhere* you added `HOST_DARWIN`.